packages/fast-components-styles-msft/src/utilities/color/neutral-layer.ts

```diff
@@ -39,14 +40,77 @@ export enum NeutralPaletteDarkModeLayers {
 }
 
 /**
- * Light mode L2 is significant because it happens at the same point as the neutral fill flip.
+ * Recommended values for light and dark mode for `baseLayerLuminance` in the design system.
+ */
+export enum StandardLuminance {
+    LightMode = 1,
+    DarkMode = 0.23,
+}
+
+function luminanceOrBackgroundColor(
+    luminanceRecipe: DesignSystemResolver<string>,
+    backgroundRecipe: DesignSystemResolver<string>
+): DesignSystemResolver<string> {
+    return (designSystem: DesignSystem): string => {
+        return baseLayerLuminance(designSystem) === -1
+            ? backgroundRecipe(designSystem)
+            : luminanceRecipe(designSystem);
+    };
+}
+
+/**
+ * Find the palette color that's closest to the desired base layer luminance.
+ */
+const baseLayerLuminanceSwatch: DesignSystemResolver<Swatch> = (
+    designSystem: DesignSystem
+): Swatch =>
+    new ColorRGBA64(
+        baseLayerLuminance(designSystem),
+        baseLayerLuminance(designSystem),
+        baseLayerLuminance(designSystem),
+        1
+    ).toStringHexRGB();
+
+/**
+ * Get the index of the base layer palette color.
+ */
+const baseLayerLuminanceIndex: DesignSystemResolver<number> = (
+    designSystem: DesignSystem
+): number =>
+    findClosestSwatchIndex(neutralPalette, baseLayerLuminanceSwatch)(designSystem);
+
+/**
+ * Get the actual value of the card layer index, clamped so we can use it to base other layers from.
+ */
+const neutralLayerCardIndex: DesignSystemResolver<number> = (
+    designSystem: DesignSystem
```

Comment:
Is this working correctly? you never access the neutralPalette from the designSystem in this function scope, so you're probably accessing the length property of the function itself

Reply:
Good catch.